File: app/src/main/java/com/druger/firstandroidgame/objects/TopBorder.java
package com.druger.firstandroidgame.objects;

import android.graphics.Bitmap;
import android.graphics.Canvas;

import com.druger.firstandroidgame.view.GamePanel;

/**
 * Created by druger on 12.08.2015.
 */
public class TopBorder extends GameObject {
    private Bitmap image;

    public TopBorder(Bitmap res, int x, int y, int height) {
        width = 20;
        this.height = height;

        this.x = x;
        this.y = y;

        dx = GamePanel.MOVESPEED;
        image = Bitmap.createBitmap(res, 0, 0, width, height);
    }

    public void update(){
        x += dx;
    }

    public void draw(Canvas canvas){
        try {
            canvas.drawBitmap(image, x, y, null);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}

File: app/src/main/java/com/druger/firstandroidgame/objects/TopBorder.kt
package com.druger.firstandroidgame.objects;

import android.graphics.Bitmap
import android.graphics.Canvas
import com.druger.firstandroidgame.view.GamePanel

/**
 * Created by druger on 12.08.2015.
 */
class TopBorder(res: Bitmap, x: Int, y: Int, height: Int) : GameObject() {
    private var image: Bitmap

    init {
        width = 20;
        this.height = height;

        this.x = x;
        this.y = y;

        dx = GamePanel.MOVESPEED;
        image = Bitmap.createBitmap(res, 0, 0, width, height);
    }

    fun update() {
        x += dx;
    }

    fun draw(canvas: Canvas) {
        try {
            canvas.drawBitmap(image, x.toFloat(), y.toFloat(), null);
        } catch (e: Exception) {
            e.printStackTrace();
        }
    }
}
